talisker/celery.py

```diff
@@ -56,10 +62,65 @@ def delay(task, *args, **kwargs):
     return task.delay(*args, **kwargs)
 
 
+# celery helpers
+def _timer(task_name, name, value):
+    name = 'celery.{}.{}'.format(task_name, name)
+    talisker.statsd.get_client().timing(name, value)
+
+
+def _counter(name):
+    def signal(sender, **kwargs):
+        stat_name = 'celery.{}.{}'.format(sender.name, name)
+        talisker.statsd.get_client().incr(stat_name)
+    return signal
+
+
+### celery signals for metrics ###
+
+def before_task_publish(sender, body, **kwargs):
+    # TODO: find a way to avoid thread locals
+    if not hasattr(_local, 'timers'):
+        _local.timers = {}
+    _local.timers[body['id']] = time.time()
+
+
+def after_task_publish(sender, body, **kwargs):
+    _timer(sender, 'enqueue', (time.time() - _local.timers.pop(body['id'])) * 1000)
+
+
+def task_prerun(sender, task_id, task, **kwargs):
+    task.__talisker_timer = time.time()
+
+
+def task_postrun(sender, task_id, task, **kwargs):
+    _timer(sender.name, 'run', (time.time() - task.__talisker_timer) * 1000)
+
+
+def enable_metrics():
+    """Best effort enabling of celery metrics"""
+    try:
+        from celery import signals
+    except ImportError:
```

Comment:
I'm wondering if this should at least emit a `NOTICE`?
Silent import swallowing, although yes the user should have celery installed if they expect celery metrics to work, but what if the signals module changes in some way or there's an `ImportError` raised within `celery.signals`?